R/estimateDiversity.R

```diff
@@ -199,15 +223,31 @@ setMethod("estimateRichness", signature = c(x = "SummarizedExperiment"),
     ans
 }
 
+.get_coverage <- function(x, threshold, ...){
+
+    # Convert table to relative values
+    otu <- .calc_rel_abund(x)
+
+    # Number of groups needed to have threshold (e.g. 50 %) of the ecosystem occupied
+    do <- apply(otu, 2, function(x) {
+        min(which(cumsum(rev(sort(x/sum(x)))) >= threshold))
+    })
+    names(do) <- colnames(otu)
+
+    do
+
+}
+
 #' @importFrom SummarizedExperiment assay assays
-.run_dvrsty <- function(x, i, mat, ...){
+.run_dvrsty <- function(x, i, mat, threshold, ...){
     dvrsty_FUN <- switch(i,
                          shannon = .get_shannon,
                          simpson = .get_simpson,
                          inv_simpson = .get_inverse_simpson,
                          richness = .get_observed,
                          chao1 = .get_chao1,
-                         ACE = .get_ACE)
-    dvrsty <- dvrsty_FUN(mat, ...)
+                         ACE = .get_ACE,
+                         coverage = .get_coverage)
+    dvrsty <- dvrsty_FUN(mat, threshold = threshold, ...)
```

Comment:
Remove the optional argument and let the `...` do their work

Reply:
Removed